includes/core/functions.php

```diff
@@ -1601,6 +1601,64 @@ function wct_is_signup_allowed_for_current_blog() {
 	return apply_filters( 'wct_is_signup_allowed_for_current_blog', wct_allow_signups() );
 }
 
+/**
+ * Updates last_status post meta before saving talk.
+ *
+ * @since  1.2.0
+ */
+function wct_talks_update_last_status( $data, $postarr ) { 
+	// Checks if we are in 'talks' CPT and if not, abort
+	if( $data['post_type'] !== 'talks' ) :
+		return $data;
+	endif;
+	// Gets current talk status
+	$current_post_status = get_post_status( $postarr['ID'] );
+	// Updates last talk status meta
+	update_post_meta( $postarr['ID'], '_wct_talks_last_status', $current_post_status );
+	return $data;
+}
```

Comment:
Yah you're all right to do it this way. I wasn't much wondering about code formatting since I was sure you'll not use my snippets in the way I made my PR :)